### PR TITLE
SpringCoolCompany

### PR DESCRIPTION
**Pull request by Dinexpod**

### CODE REVIEW

File: src/main/webapp/WEB-INF/pages/developer/jqueryDev.jsp [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+<button type="button" name="back" onclick="history.back()">back</button>
+
+<div id="allDevelopers">
+
+</div>
+
+
+<script
+        src="https://code.jquery.com/jquery-3.3.1.js"
```
Comment (Gzyablik) [inline src/main/webapp/WEB-INF/pages/developer/jqueryDev.jsp]: It's not good practice to use outer dependencies from web. What will you do if the servers of jquery.com will burn in flame? You can keep it in your project.

File: src/main/java/mate/academy/spring/model/Developer.java [hunk 2]
```diff
@@ -19,7 +23,15 @@
...
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long developerId;
+    private String developerName;
     private Integer developerAge;
     private Integer developerSalary;
-    private String developerName;
+
+    @ManyToMany
+    @JoinTable(name="developer_project")
+    private Set<Project> developerProjects = new HashSet<>();
```
[the remark this answers is not in the document]
Comment (Dinexpod) [reply src/main/java/mate/academy/spring/model/Developer.java]: ok

File: src/main/java/mate/academy/spring/controller/CompanyController.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    @Autowired
+    public CompanyController(CompanyService companyService) {
+        this.companyService = companyService;
+    }
+
+    @RequestMapping(method = RequestMethod.GET)
+    public String company() {
+        System.out.println("controller: COMPANY");
+//        companyService.create();
```
[the remark this answers is not in the document]
Comment (Dinexpod) [reply src/main/java/mate/academy/spring/controller/CompanyController.java]: ok

File: src/main/java/mate/academy/spring/service/CompanyServiceImpl.java [hunk 4]
```diff
@@ -18,13 +20,21 @@ public CompanyServiceImpl(CompanyRepository companyRepository) {
...
     }
 
     @Override
     public Company getCompany(Long id) {
         return null;
     }
+
+    @PostConstruct
+    public void postConstruct() {
+        System.out.println("CompanyServiceImpl is OK!");
```
[the remark this answers is not in the document]
Comment (Dinexpod) [reply src/main/java/mate/academy/spring/service/CompanyServiceImpl.java]: ok

File: src/main/java/mate/academy/spring/service/DeveloperService.java [hunk 5]
```diff
@@ -1,8 +1,18 @@
...
+
 public interface DeveloperService {
     void create();
-    Developer getDeveloper(Long id);
+
+    Developer create(Developer developer);
+
+    DeveloperDto getDeveloper(Long id);
+
+    List<Developer> getAll();
```
[the remark this answers is not in the document]
Comment (Dinexpod) [reply src/main/java/mate/academy/spring/service/DeveloperService.java]: ok

File: src/main/java/mate/academy/spring/controller/AuthController.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+    @RequestMapping(value = "/reg", method = RequestMethod.GET)
+    public String showAuth() {
+
+        return "auth/reg";
+    }
+
+    @RequestMapping(value = "/login", method = RequestMethod.POST)
+    public String login(UserLoginInput user) {
+        System.out.println("LOGIIIIIIIIIIIIIIIIIIIIN");
```
[the remark this answers is not in the document]
Comment (Dinexpod) [reply src/main/java/mate/academy/spring/controller/AuthController.java]: ok

File: pom.xml [hunk 7]
```diff
@@ -82,6 +82,20 @@
...
+        <!-- https://mvnrepository.com/artifact/log4j/log4j -->
+        <dependency>
+            <groupId>log4j</groupId>
+            <artifactId>log4j</artifactId>
+            <version>1.2.17</version>
+        </dependency>
+        <!-- https://mvnrepository.com/artifact/org.webjars/jquery -->
+        <dependency>
+            <groupId>org.webjars</groupId>
+            <artifactId>jquery</artifactId>
```
Comment (boroda4436) [inline pom.xml]: ?????
Why you need this?

Comment (Dinexpod) [reply pom.xml]: ow! I foget this, it`s my mistakes, last week I thought that it needs.

**Pull request merged**